src/compile/legend/encode.ts

```diff
@@ -112,3 +112,13 @@ export function labels(fieldDef: FieldDef<string>, labelsSpec: any, model: UnitM
   return keys(labels).length > 0 ? labels : undefined;
 }
 
+function getOpacityValue(opacityDef: ConditionalFieldDef<LegendFieldDef<string>> | ConditionalValueDef<LegendFieldDef<string>>): number {
+  if (isValueDef(opacityDef)) {
+    if (hasConditionValueDef(opacityDef)) {
```

Comment:
wonder if we can just do 
```
if (opacityDef.condition && isValueDef(opacityDef))
```

so we do not have to increase the number of type guards